src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -568,7 +568,7 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                 result.WasCompilerGenerated = true;
                 diagnostics.AddRange(bag);
 
-                if (bag.HasAnyErrors())
+                if (bag.HasAnyResolvedErrors())
                 {
```

Comment:
The original comment this PR intends to address isn't really to the point. The circularities the lazy error mechanism is needed to avoid are in member declaration processing. Elsewhere in the compiler, where we cannot be processing member declarations, we can ignore whether or not errors are resolved. This code path doesn't occur in member declaration processing, and so the resolution status can be ignored.

More to the point, we should not be looking in the bag for errors to decide if we've found a suitable deconstruct method or not. Whatever errors we have here can simple be relayed to the result if the shape is otherwise correct.